apps/common/src/python/mediawords/solr/request.py

```diff
@@ -152,6 +157,54 @@ def __solr_error_message_from_response(response: Response) -> str:
     return error_message
 
 
+def merge_responses(dict1: dict,dict2: dict):
```

Comment:
This looks good @thepsalmist I haven't tested the code below but I would suggest we try something like this; I think it reduces some repetition. cc @kilemensi

```
def merge_responses(mc_32_bit_collection: dict,mc_64_bit_collection: dict):
    new_response = {}

    new_response.update(mc_32_bit_collection.get("responseHeader", {})) # Will prevent a KeyError

    mc_32_bit_response = mc_32_bit_collection.get("response", {})
    mc_64_bit_response = mc_64_bit_collection.get("response", {})

    num_found = mc_32_bit_response.get("numFound", 0) + mc_64_bit_response.get("numFound", 0)
    start_index = mc_32_bit_response.get("start", 0) + mc_64_bit_response.get("start", 0)

    docs = []

    docs.extend(mc_32_bit_response.get("docs", [])).extend(mc_64_bit_response.get("docs", []))

    new_response.update({
        "response": {
            "numFound": num_found,
            "start": start_index,
            "docs": docs,
        }
    })

    # facets
    mc_32_bit_facets = mc_32_bit_response.get("facets", {})
    mc_64_bit_facets = mc_64_bit_response.get("facets", {})

    count = mc_32_bit_facets.get("count", 0) + mc_64_bit_facets.get("count", 0)

    categories = {}

    if 'categories' in mc_32_bit_facets:
        if mc_32_bit_facets.get('categories', {}).get('buckets', []):
            categories.update({"buckets": mc_32_bit_facets.get('categories', {}).get('buckets', []).extend(mc_64_bit_facets.get('categories', {}).get('buckets', []))})

    new_response.update({
        "facets": {
            "count": count,
            "categories": categories,
        }
    })

    return new_response
```

Reply:
LGTM @esirK